### ISSUE TITLE
Xcode 8 / Swift 3 compatibility

### ISSUE THREAD

**Issue by lfarah** [Flinesoft/BartyCrouch#33]
Hey, I'd love to see this lib available in Swift 3 so I can implement it in my project.

Thanks 😄

**Comment by Dschee** [comment]
I just added support for Swift 3, but unfortunately the Swift compiler throws an error when installing it:

```
!257 = <temporary!> !DICompositeType(tag: DW_TAG_subroutine_type, name: "main", scope: !5, file: !5, flags: DIFlagFwdDecl)
Expected no forward declarations!
!257 = <temporary!> !DICompositeType(tag: DW_TAG_subroutine_type, name: "main", scope: !5, file: !5, flags: DIFlagFwdDecl)
LLVM ERROR: Broken function found, compilation aborted!
2016-09-22 01:18:38.385 xcodebuild[14131:14406770]  DVTAssertions: Warning in /Library/Caches/com.apple.xbs/Sources/IDEFrameworks/IDEFrameworks-11246/IDEFoundation/Playgrounds/IDEPlaygroundAuxiliarySourceCompilerOperation.m:386
Details:  Unable to read diagnostics from file "/Users/Me/Library/Developer/Xcode/DerivedData/BartyCrouch-gtgqbgwlukpdggfcuyrcjgxeurwn/Build/Intermediates/ArchiveIntermediates/BartyCrouch CLI/IntermediateBuildFilesPath/BartyCrouch.build/Release/BartyCrouch CLI.build/Objects-normal/x86_64/CommandLineKit.dia" (Invalid File): Invalid diagnostics signature
Function: void XCGenerateDiagnosticsFromFile(NSString *__strong, NSString *__strong, NSDictionary *__strong, NSDictionary *__strong, IDEActivityLogSectionRecorder *__strong, BOOL (^__strong)(IDEActivityLogMessage *__strong))
Thread:   <NSThread: 0x7fedefaf9c70>{number = 16, name = (null)}
Please file a bug at http://bugreport.apple.com with this warning message and any useful information you can provide.

Command /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/swiftc failed with exit code 1
** INSTALL FAILED **


The following build commands failed:
	CompileSwift normal x86_64
	CompileSwiftSources normal x86_64 com.apple.xcode.tools.swift.compiler
(2 failures)
```

I'm filing a bug report to the Swift team as I think this is a Swift bug. It's building completely fine in Xcode ...

If you (or anybody else) have an idea how to work around this, I'm happily willing to change the code. Version 3.3.2 (see Tags) is the version that should have support for Swift 3.

**Comment by Dschee** [comment]
I filed a bug report on the Swift issue tracker here:
https://bugs.swift.org/browse/SR-2725

**Comment by Dschee** [comment]
I fixed this with the hint posted in https://bugs.swift.org/browse/SR-2725 and released version 3.3.3! 🎉
